File: System.Windows.Forms.Metro/System.Windows.Forms.Metro/ToolTip.vb
Imports System.ComponentModel

Namespace System.Windows.Forms.Metro

    <ToolboxBitmap(GetType(System.Windows.Forms.ToolTip))> _
    Public Class ToolTip
        Inherits System.Windows.Forms.ToolTip

        Public Sub New()
            Me.InitialDelay = 1000
            Me.BackColor = System.Drawing.Color.Black
            Me.ForeColor = Color.White
            Me.UseFading = True
            Me.UseAnimation = True
            Me.OwnerDraw = True
            Me.AutomaticDelay = 500
            Me.AutoPopDelay = 2000
        End Sub

        Private Sub ToolTip_Draw(sender As Object, e As DrawToolTipEventArgs) Handles Me.Draw
            e.DrawBackground()
            e.DrawText() 'TextFormatFlags.WordBreak)
        End Sub

        Private Sub ToolTip_Popup(sender As Object, e As PopupEventArgs) Handles Me.Popup
            If Me.BackColor.GetBrightness >= 0.5 Then Me.ForeColor = Color.White
        End Sub

    End Class

End Namespace

File: TodoList/Controls/ToolTip.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace Xuld.TodoList.Controls {

    [ToolboxBitmap(typeof(System.Windows.Forms.ToolTip))]
    public class ToolTip : System.Windows.Forms.ToolTip {

        public ToolTip() {
            InitialDelay = 1000;
            BackColor = System.Drawing.Color.Black;
            ForeColor = Color.White;
            UseFading = true;
            UseAnimation = true;
            OwnerDraw = true;
            AutomaticDelay = 500;
            AutoPopDelay = 2000;

            Draw += ToolTip_Draw;
            Popup += ToolTip_Popup;

        }

        void ToolTip_Popup(object sender, System.Windows.Forms.PopupEventArgs e) {
            if (BackColor.GetBrightness() >= 0.5) {
                ForeColor = Color.White;
            }
        }

        void ToolTip_Draw(object sender, System.Windows.Forms.DrawToolTipEventArgs e) {
            e.DrawBackground();
            e.DrawText();
        }

    }
}
